podcast/2.13.md

```diff
@@ -0,0 +1,913 @@
+
+If you're young in finance, you basically become a programmer just because the older people don't want to learn it so they throw all the coding related stuff to the juniors. And of course they don't have a budget to hire a dedicated programmer. So I started with VBA just to automate Excel and then in Excel you hit some limits and it's a very crude language so then you go to Python and then I was doing currency derivatives trading. So I did Haskell and people are always impressed that I learned Haskell and did production Haskell for two years.
+
+And I didn't consider myself a programmer just because I didn't have a choice that the house language, the house library for calculating option pricing, was in Haskell. It was pretty funny because I think basically one guy got a job there and then just pulled in the entire Haskell core team to work there. And at some point, like instead of trying, the whole bank was just pot committed. They had built so much infrastructure and then they just couldn't. So it’s pretty funny how you embed a language community within a company and kind of don't give them a choice. This very similar thing that Jordan Walke did at Reason for ReasonML and Facebook. Now they're called Rescript because if you just use it to build production infrastructure, then they now rely on it.
+
+And that's a very fun trick to grow a language. Anyway, I never considered, and just to be clear, like I did this, I built tools that I could use for myself, right. So I did not consider myself a programmer. Although I could code, it was like 5% of my job rather than like 90% of my job, right. So that's the way that I would frame it. And eventually, I think what I realized was that I was a script monkey. I would run analysis, do a lot of number crunching and then my boss would be like, “But what about this other scenario?” And then I would have to go do that, go do a different scenario again. And I realized that I was kind of the bottleneck for my code no matter how well I did. I was the sort of data analyst, so punching away at the thing. And I wanted to do user interfaces–that's the real unlock for turning business logic into products, essentially, that anyone could use.
+
+So then I pivoted hard into JavaScript and that was actually harder to learn than Haskell, I would say, because it's a very incompatible and confusing ecosystem. Haskell has a very small user base. Everything is functionally typed. There's very strong guarantees. JavaScript is not that, right? Like when I was learning it, it was still like, is this es5? Like we're going into es6. But you transpile this here and then like there's node, that's different from the browser stuff.
```

Comment:
```suggestion
So then I pivoted hard into JavaScript and that was actually harder to learn than Haskell, I would say, because it's a very incompatible and confusing ecosystem. Haskell has a very small user base. Everything is functionally typed. There's very strong guarantees. JavaScript is not that, right? Like when I was learning it, it was still like, is this ES5? Like we're going into ES6. But you transpile this here and then like there's Node, that's different from the browser stuff.
```